tools/amp_stt/tag_words.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+import argparse
+import csv
+from pathlib import Path
+import json
+import logging
+import sys
+import os
+sys.path.insert(0, os.path.abspath('../../../../../tools/amp_schema'))
+from speech_to_text import SpeechToText
+def main():
+    amp_transcript =  sys.argv[1] 
+    words_to_flag_file =  sys.argv[2] 
+    output_csv =  sys.argv[3] 
+
+    # Get a list of words to flag
+    words_to_flag = get_words(words_to_flag_file) 
+    print("Words to Flag:")
+    for w in words_to_flag:
+        print(w)
+    print("")
+
+    # Search for matching words/phrases
+    matching_words = list()
+    with open(amp_transcript, 'r') as f:
+        transcript_dict = json.load(f)
+        # Check to see if we have a valid transcript.  If so, find the matches. 
+        if "results" in transcript_dict.keys() and "words" in transcript_dict["results"].keys():
+            transcript_words = transcript_dict["results"]["words"]
+            matching_words = match_words(words_to_flag, transcript_words)
+
+    # Print the output
+    print(matching_words)
+    write_csv(output_csv, matching_words)
+    exit(0)
+
+def match_words(words_to_flag, transcript_words):
+    matching_words = list()
+    # Iterate through the transcript
+    for word in transcript_words:
+        cleaned_word = clean_word(word["text"])
```

Comment:
I'm pretty sure the transcript words have no space around, but it does hurt to clean_word on it.

Reply:
Does or does not hurt?  Clean turns it to lower case for comparison purposes too.

Reply:
I thought about the boundary and punctuation question...because I don't filter on punctuation, it acts as it's own boundary.  If I remove the punctuation, "potato sack" and "potato, sack" will match.  But leaving it in the word list keeps the difference between the two.

Reply:
Yes, you are right! I added that "on a second thought". 

One more thought, also a minor corner case, but doesn't hurt to clean that out: we can filter out redundant phrases (after cleaning) among the tagging words; otherwise we can end up doing matching twice on them. if you use hashmap to store phrases (or parts) that can easily do.

Reply:
Good idea.  I filtered out redundant phrases with a dict.

Reply:
Also added filtering for certain punctuation in flag words (!,.).  Seems to work, we still find "you've", but don't match the "potato, sack" issue.